lib/states/il/chicago.R

```diff
@@ -57,42 +80,102 @@ clean <- function(d, helpers) {
   )
 
   d$data %>%
-    rename(
-      subject_age = age,
-      violation = statute_description,
-			officer_id = officer_employee_no
+    # date, time,
+    separate(contactdate, into=c("date_new","time_new"), sep=" ") %>%
+    mutate(
+      date = if_else(
+        source == "old_data",
+        date(ymd(date)),
+        date(dmy(date_new))
+      ),
+      time = if_else(
+        source == "old_data",
+        parse_time(arrest_hour, "%H"),
+        parse_time(time_new)
+      )
     ) %>%
+    # location, county, beat, district,
     mutate(
-      type = if_else(
-        str_detect(violation, "PEDEST")
-        & !str_detect(violation, "FAILURE TO YIELD TO PEDESTRIAN")
-        & !str_detect(violation, "PASS VEH STOPPED FOR PEDEST"),
-        "pedestrian",
-        "vehicular"
+      location = if_else(
+        source == "old_data",
+        str_c(street_no, street_name, street_direction, sep = " "),
+        str_c(street_no, dir, street_nme, sep = " ")
       ),
-      time = parse_time(arrest_hour, "%H"),
-      location = str_trim(
-        str_c(
-          street_no,
-          street_name,
-          street_direction,
-          sep = " "
-        )
+      county = "Cook County"
+    ) %>%
+    # subject_age, subject_race, subject_sex
+    mutate(
+      subject_age = if_else(
+        source == "old_data",
+        as.numeric(age),
+        year(date) - as.numeric(year_of_birth)
       ),
-      subject_race = fast_tr(coalesce(race, driver_race), tr_race),
-      subject_sex = fast_tr(coalesce(gender, driver_gender), tr_sex),
-      officer_race = fast_tr(officer_race, tr_race),
-      officer_sex = fast_tr(officer_gender, tr_sex),
-      outcome = first_of(
-        arrest = arrest_made,
-        citation = citation_issued
+      subject_race = if_else(
+        source == "old_data",
+        fast_tr(coalesce(race, driver_race), tr_race),
+        fast_tr(race, tr_race)
+      ),
+      subject_sex = if_else(
+        source == "old_data",
+        fast_tr(coalesce(gender, driver_gender), tr_sex),
+        fast_tr(sex, tr_sex)
       )
     ) %>%
-    helpers$add_lat_lng(
+    # officer vars
+    rename(officer_id = officer_employee_no) %>%
+    mutate(
+      officer_race = fast_tr(officer_race, tr_race),
+      officer_sex = fast_tr(officer_gender, tr_sex)
     ) %>%
+    # unit, type, violation
+    rename(unit = cpd_unit_no) %>%
+    mutate(
+      violation = if_else(source == "old_data", statute_description, statute),
+      type = if_else(source == "old_data",
+                     if_else(
+                       str_detect(violation, "PEDEST")
+                       & !str_detect(violation, "FAILURE TO YIELD TO PEDESTRIAN")
+                       & !str_detect(violation, "PASS VEH STOPPED FOR PEDEST"),
+                       "pedestrian",
+                       "vehicular"
+                     ),
+                     "vehicular")
```

Comment:
The whitespace here is a little weird which makes it hard to read. I'd stick with two spaces for each level of indentation. Also, can you add a short comment above explaining the violation type imputation logic?